src/Kentico.Xperience.Algolia/Admin/AlgoliaModuleInstaller.cs

```diff
@@ -0,0 +1,302 @@
+﻿using CMS.DataEngine;
+using CMS.FormEngine;
+using CMS.Modules;
+
+namespace Kentico.Xperience.Algolia.Admin;
+
+internal class AlgoliaModuleInstaller
+{
+    private readonly IResourceInfoProvider resourceProvider;
+
+    public AlgoliaModuleInstaller(IResourceInfoProvider resourceProvider) =>
+        this.resourceProvider = resourceProvider;
+
+    public void Install()
+    {
+        var resource = InstallResource();
+
+        InstallAlgoliaItemInfo(resource);
+        InstallAlgoliaLanguageItemInfo(resource);
+        InstallAlgoliaIndexPathItemInfo(resource);
+        InstallAlgoliaContentTypeItemInfo(resource);
+    }
+
+    private ResourceInfo InstallResource()
+    {
+        var resourceInfo = resourceProvider.Get("CMS.Integration.Algolia") ?? new ResourceInfo();
+
+        resourceInfo.ResourceDisplayName = "Algolia Search";
+        resourceInfo.ResourceName = "Kentico.Xperience.Algolia";
+        resourceInfo.ResourceDescription = "Kentico Algolia custom data";
+        resourceInfo.ResourceIsInDevelopment = false;
+        if (resourceInfo.HasChanged)
+        {
+            resourceProvider.Set(resourceInfo);
+        }
+
+        return resourceInfo;
+    }
+
+    private static void InstallAlgoliaItemInfo(ResourceInfo resource)
+    {
+        var algoliaItemInfo = DataClassInfoProvider.GetDataClassInfo(AlgoliaIndexItemInfo.OBJECT_TYPE);
```

Comment:
Take a look at how we're doing this in Lucene to minimize disruption to the CI repository:

https://github.com/Kentico/xperience-by-kentico-lucene/blob/main/src/Kentico.Xperience.Lucene/Admin/LuceneModuleInstaller.cs#L44
https://github.com/Kentico/xperience-by-kentico-lucene/blob/main/src/Kentico.Xperience.Lucene/Admin/LuceneModuleInstaller.cs#L113
https://github.com/Kentico/xperience-by-kentico-lucene/blob/main/src/Kentico.Xperience.Lucene/Admin/LuceneModuleInstaller.cs#L115

Reply:
changed